From 4ea50aa41ff0131b380aef97623be578b702cac7 Mon Sep 17 00:00:00 2001
From: Mart Raudsepp <mart@leio.tech>
Date: Fri, 24 Sep 2021 00:02:40 +0300
Subject: [PATCH] build-sys: meson: Relax sanity checks for daemonless builds

Echo canceller, udev and speex resampling is only used by the daemon, so
don't error or warn about those in -Ddaemon=false builds.
---
 meson.build | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/meson.build b/meson.build
index d7e468cab..63ed62342 100644
--- a/meson.build
+++ b/meson.build
@@ -961,7 +961,7 @@ message('\n    '.join(summary))
 
 # Sanity checks
 
-if not speex_dep.found() and not webrtc_dep.found() and not get_option('adrian-aec')
+if get_option('daemon') and not speex_dep.found() and not webrtc_dep.found() and not get_option('adrian-aec')
   error('At least one echo canceller implementation must be available!')
 endif
 
@@ -981,7 +981,7 @@ if host_machine.system() != 'windows'
     ]
     warning('\n' + '\n'.join(message))
   endif
-  if host_machine.system() == 'linux' and not udev_dep.found()
+  if host_machine.system() == 'linux' and not udev_dep.found() and get_option('daemon')
     message = [
       'You do not have udev support enabled. It is strongly recommended',
       'that you enable udev support if your platform supports it as it is',
@@ -990,7 +990,7 @@ if host_machine.system() != 'windows'
     ]
     warning('\n' + '\n'.join(message))
   endif
-  if not speex_dep.found()
+  if not speex_dep.found() and get_option('daemon')
     message = [
       'You do not have speex support enabled. It is strongly recommended',
       'that you enable speex support if your platform supports it as it is',
-- 
2.32.0

